sources/AppBundle/Security/MyGithubAuthenticator.php

```diff
@@ -18,9 +18,11 @@
 use Symfony\Component\Routing\RouterInterface;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
-use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+use Symfony\Component\Security\Http\EntryPoint\AuthenticationEntryPointInterface;
 
-class MyGithubAuthenticator extends SocialAuthenticator
+class MyGithubAuthenticator extends OAuth2Authenticator implements AuthenticationEntryPointInterface
```

Comment:
Ce nom de classe on dirait un truc de tuto. On pourrait en profiter pour le changer :

```suggestion
class GithubAuthenticator extends OAuth2Authenticator implements AuthenticationEntryPointInterface
```

Reply:
Oui, c'est plus propre en effet.